analyses/new-feature-template.qmd

```diff
@@ -0,0 +1,1123 @@
+date: "`r Sys.Date()`"
+author: "Cook County Assessor's Office Data Department"
+execute:
+  echo: false
+  warning: false
+  asis: true
+format:
+  html:
+    embed-resources: true
+    toc: true
+    toc_float: true
+    fig-align: center
+    fontsize: 12pt
+params:
+  run_id: "2024-07-03-charming-boni"
+  run_id_year: "2024"
+  comparison_run_id: "2024-07-09-cool-takuya"
+  comparison_run_id_year: "2024"
+  added_feature: "prox_nearest_new_construction_dist_ft" 
+  added_feature_shap: "prox_nearest_new_construction_dist_ft_shap"
+  description: "A distance in feet to the nearest new construction"
+  min_range: 5
+  max_range: 95
+  type: "continuous"
+---
+
+```{r packages}
+library(purrr)
+library(here)
+library(yaml)
+library(corrplot)
+# Load list of helper files and main libraries
+purrr::walk(list.files(here::here("R"), "\\.R$", full.names = TRUE), source)
+
+# Load reporting-only R libraries
+suppressPackageStartupMessages({
+  reporting_libs <- "Config/renv/profiles/reporting/dependencies"
+  purrr::walk(
+    strsplit(read_yaml(here::here("DESCRIPTION"))[[reporting_libs]], ", ")[[1]],
+    library,
+    character.only = TRUE
+  )
+})
+
+# TODO: Catch for weird Arrow bug with SIGPIPE. Need to permanently fix later
+# https://github.com/apache/arrow/issues/32026
+cpp11::cpp_source(code = "
+#include <csignal>
+#include <cpp11.hpp>
+
+[[cpp11::register]] void ignore_sigpipes() {
+  signal(SIGPIPE, SIG_IGN);
+}
+")
+
+ignore_sigpipes()
+```
+
+```{r download_new_data}
+base_paths <- model_file_dict(params$run_id, params$run_id_year)
+comparison_paths <- model_file_dict(
+  params$comparison_run_id,
+  params$comparison_run_id_year
+)
+run_id <- params$run_id
+comparison_run_id <- params$comparison_run_id
+
+analyses_paths <- list(
+  output = list(
+    list(
+      s3 = base_paths$output$assessment_card$s3,
+      key = "assessment_card"
+    ),
+    list(
+      s3 = base_paths$output$assessment_pin$s3,
+      key = "assessment_pin"
+    ),
+    list(
+      s3 = base_paths$output$metadata$s3,
+      key = "metadata"
+    ),
+    list(
+      s3 = base_paths$output$performance_test$s3,
+      key = "performance_test"
+    ),
+    list(
+      s3 = base_paths$output$shap$s3,
+      key = "shap"
+    )
+  )
+)
+
+source("Analysis_helpers.R")
+data_new <- model_fetch_run_subset(
+  params$run_id,
+  params$run_id_year,
+  analyses_paths, TRUE
+)
+
+list2env(data_new, envir = .GlobalEnv)
+
+rm(data_new)
+
+comparison_paths <- list(
+  output = list(
+    list(
+      s3 = base_paths$output$assessment_pin$s3,
+      key = "assessment_pin"
+    ),
+    list(
+      s3 = base_paths$output$performance_test$s3,
+      key = "performance_test"
+    )
+  )
+)
+
+data_comparison <- model_fetch_run_subset(
+  params$comparison_run_id,
+  params$comparison_run_id_year,
+  comparison_paths, TRUE
+)
+
+list2env(data_comparison, envir = .GlobalEnv)
+
+rm(data_comparison)
+
+all_vars <- ls()
+
```

Comment:
This is in the helper file, where we rename the appended values to _new and _comp to make the coding easier to read.